APIAlfresco.php

```diff
@@ -1,475 +1,552 @@
-<?php 
-/*
- *		@author: Daniel Ojeda
- * 		@Version: 1.2
+<?php
+
+/**
+ * @author: Daniel Ojeda
+ * @Version: 1.2
  */
 
-/*
-* 		@usage:
-*			$conexion = APIAlfresco::getInstance();
-*			$conexion->connect($urlRepository,$User,$Pass);
-*			$conexion->checkResponse();
-*			$conexion->setFolderByPath($folder);
-*/
+/**
+ * @usage:
+ * $conexion = APIAlfresco::getInstance();
+ * $conexion->connect($urlRepository,$User,$Pass);
+ * $conexion->checkResponse();
+ * $conexion->setFolderByPath($folder);
+ */
+require_once 'cmis_repository_wrapper.php';
+require_once 'cmis_service.php';
+
+class APIAlfresco
+{
+    private static $instance;
+    public $urlRepository;
+    public $user;
+    public $pass;
+    public $repository;
+    public $parentFolder;
+
+    // Singleton
+
+    public function __construct()
+    {
+    }
 
-require_once ("cmis_repository_wrapper.php");
-require_once ("cmis_service.php");
+    public static function getInstance()
+    {
+        if (!isset(self::$instance)) {
+            $obj = __CLASS__;
+            self::$instance = new $obj();
+        }
 
-class APIAlfresco{
+        return self::$instance;
+    }
 
-	private static $instance;
-	public $urlRepository;
-	public $user;
-	public $pass;
-	public $repository;
-	public $parentFolder;
+    public function __clone()
+    {
+        trigger_error('Clone not allowed.', E_USER_ERROR);
+    }
 
-	// Singleton
+    /**
+     * Function to connect to alfresco.
+     *
+     * @example $url = "http://127.0.0.1:8080/alfresco/cmisatom";
+     *
+     * @param string $url
+     * @param string $user
+     * @param string $pass
+     */
+    public function connect($url, $user, $pass)
+    {
+        $this->urlRepository = $url;
+        $this->user = $user;
+        $this->pass = $pass;
+        $this->repository = new CMISService($url, $user, $pass);
+    }
 
-	function __construct()
-	{
-		
-	}
+    public function checkResponse()
+    {
+        if ($this->repository->getLastRequest()->code > 299) {
```

Comment:
Add this const to the class i.e:

`const ERROR_CODE_REASON_X = 299`

and then in your condition

`if ($this->repository->getLastRequest()->code > self::ERROR_CODE_REASON_X)`